rt/frontend/jaxrs/src/main/java/org/apache/cxf/jaxrs/utils/InjectionUtils.java

```diff
@@ -1167,10 +1161,12 @@ public static void injectContextMethods(Object requestObject,
                 if (!cri.isSingleton()) {
                     InjectionUtils.injectThroughMethod(requestObject, method, o, message);
                 } else {
-                    ThreadLocalProxy<Object> proxy 
-                        = (ThreadLocalProxy<Object>)cri.getContextSetterProxy(method);
+                    Object proxy = extractFromSetter(requestObject, method);
+                    if (!(proxy instanceof ThreadLocalProxy)) {
```

Comment:
Np, thanks. See if we see an unexpected value then it means there's a problem lying somewhere else, this is where my concern is. If some code somewhere depends on this 'unexpected' value then the proposed fix will break that code, right ? If a bean naming pattern is not followed then IMHO CXF should not worry about it. In case of CXF-7309, what is returned there ? Thanks